### ISSUE TITLE
Randomly getting 'Email Already in use' error while trying to signin

### ISSUE THREAD

**Issue by AslamThachapalli** [serverpod/serverpod#2109]
I was trying to Sign up using serverpod_auth_email. But while pressing create account button I am getting 'Email already in user' error even though there are no entries in the db. 
Video:

https://github.com/serverpod/serverpod/assets/113328135/ac4c1dfc-dc04-4d37-b441-686b2d8dfc8e

My DB fields:
![Screenshot 2024-04-08 101416](https://github.com/serverpod/serverpod/assets/113328135/9574d9ea-2f34-4bad-be62-1fe0064b9300)

`server.dart` code: 
```dart
  auth.AuthConfig.set(
    auth.AuthConfig(
      sendValidationEmail: (session, email, validationCode) async {
        print('email validationCode: $validationCode');
        return true;
      },
      sendPasswordResetEmail: (session, userInfo, validationCode) async {
        print('password reset validationCode: $validationCode');
        return true;
      },
    ),
  );
```

**To Reproduce**
1. Email Authentication flow implementation as in documentation.
2. Using `SignInWithEmailButton` to input the field
3. Enter fields and press `Create Account` button.

**Expected behavior**
The user should be onboarded if there is no email conflict. But getting the email conflict even thought there are no users in the db.

**Serverpod versions**
 -  _CLI Version:_ 1.2.6
 - _Library version:_ 1.2.6

**Platform information**
Running on Windows 11. 
Tested on Real device.

**Comment by BenAuerDev** [comment]
Hey man, 

this is a known issue, reported in #1969 and #2051.
This is due to this part in the code (I think)

![Screenshot from 2024-04-08 09-37-28](https://github.com/serverpod/serverpod/assets/136239531/e295cf19-893d-4ef3-b533-494b6a6cf782)

which returns "Email already in use', when the authentication does not succeed. 

It's a little more tricky to set up, but it is possible. 

Did you run serverpod generate after you added the module to the server https://docs.serverpod.dev/concepts/authentication/setup#server-setup ?

**Comment by AslamThachapalli** [comment]
Yeah, @BenAuerDev 
I ran serverpod generate.

Infact I shutdown everything, created new db, applied fresh migrations, started server and tried again. But still it is giving the same issue. 

I remember last day I tried this in my Mac machine and for the first 2 tries it gave this issue, but on the third try I was able to sign in even though I didn't make any changes to the code. But today in windows, I am following the exact steps, but not able to sign in no matter how many times I retries.

**Comment by BenAuerDev** [comment]
@AslamThachapalli damn that's bitter :-/. 

Maybe try this config/project on MAC again to see if it has to do with windows

**Comment by AslamThachapalli** [comment]
Oh god. 
I got the error. It was some lack of attention from my end. 
Used wrong port to connect with the server.

I somehow happened to modify the port from 8080 to 8000, and that caused the error. 
But due to the random 'Email already in use' error, I didn't pay attention to it. 
I tried another endpoint, not the authentication one, and thus came to know about the blunder. 

**It is very much required to have a meaningful error message while authentication.**

Closing this issue.

Thanks @BenAuerDev for your responses.
